src/main/java/org/dataone/hashstore/filehashstore/FileHashStore.java

```diff
@@ -0,0 +1,1386 @@
+    private final Path OBJECT_STORE_DIRECTORY;
+    private final Path OBJECT_TMP_FILE_DIRECTORY;
+    private final String METADATA_NAMESPACE;
+    private final Path METADATA_STORE_DIRECTORY;
+    private final Path METADATA_TMP_FILE_DIRECTORY;
+
+    public static final String[] SUPPORTED_HASH_ALGORITHMS = { "MD2", "MD5", "SHA-1", "SHA-256", "SHA-384", "SHA-512",
+            "SHA-512/224", "SHA-512/256" };
+
+    enum DefaultHashAlgorithms {
+        MD5,
+        SHA_1,
+        SHA_256,
+        SHA_384,
+        SHA_512
+    }
+
+    enum HashStoreProperties {
+        storePath,
+        storeDepth,
+        storeWidth,
+        storeAlgorithm,
+        storeMetadataNamespace
+    }
+
+    /**
+     * Constructor to initialize HashStore, properties are required.
+     * 
+     * Note: HashStore is not responsible for ensuring that the given store path is
+     * accurate. It will only check for an existing configuration, directories or
+     * objects at the supplied store path before initializing.
+     * 
+     * Four directories will be created based on the given storePath string:
+     * - .../[storePath]/objects
+     * - .../[storePath]/objects/tmp
+     * - .../[storePath]/metadata
+     * - .../[storePath]/metadata/tmp
+     * 
+     * @param hashstoreProperties HashMap<String, Object> of the following keys:
+     *                            storePath (Path)
+     *                            storeDepth (int)
+     *                            storeWidth (int)
+     *                            storeAlgorithm (String)
+     *                            storeMetadataNamespace (String)
+     * @throws IllegalArgumentException Constructor arguments cannot be null, empty
+     *                                  or less than 0
+     * @throws IOException              Issue with creating directories
+     * @throws NoSuchAlgorithmException Unsupported store algorithm
+     */
+    public FileHashStore(HashMap<String, Object> hashstoreProperties)
+            throws IllegalArgumentException, IOException, NoSuchAlgorithmException {
+        if (hashstoreProperties == null) {
+            String errMsg = "FileHashStore - Properties cannot be null.";
+            logFileHashStore.fatal(errMsg);
+            throw new NullPointerException(errMsg);
+
+        }
+
+        // Get properties
+        Path storePath = (Path) hashstoreProperties.get(HashStoreProperties.storePath.name());
+        int storeDepth = (int) hashstoreProperties.get(HashStoreProperties.storeDepth.name());
+        int storeWidth = (int) hashstoreProperties.get(HashStoreProperties.storeWidth.name());
+        String storeAlgorithm = (String) hashstoreProperties.get(HashStoreProperties.storeAlgorithm.name());
+        String storeMetadataNamespace = (String) hashstoreProperties
+                .get(HashStoreProperties.storeMetadataNamespace.name());
+
+        // Validate input parameters
+        if (storePath == null) {
```

Comment:
It is not okay - in the refactor to require a `Properties` object instead of a `HashMap`, an exception will be thrown when the `storePath` variable is being declared if the supplied hashstoreProperties `storePath` value is null. A new junit test has also been added to confirm this behaviour.